### PR TITLE
feat: add full activation function options to Conv2D and Dense nodes

### LINKED ISSUE #129: Add Activation Function Options to Conv2D
**Difficulty:** Medium
**Area:** frontend

## Description

The Conv2D node currently has limited activation function options in its property editor. In practice, convolutional layers benefit from a variety of activation functions depending on the architecture and task. This issue ensures that the Conv2D layer supports a complete set of commonly used activation functions: relu, sigmoid, tanh, softmax, elu, selu, and none.

The backend already passes the activation string directly to `tf.keras.layers.Conv2D(activation=activation)`, which supports all standard Keras activation names natively. The work is primarily on the frontend: ensuring the NodePropertiesPanel presents all options in the activation dropdown when a Conv2D node is selected, and that the ConvNode component displays the selected activation correctly.

## Acceptance Criteria

- [ ] Conv2D node property panel shows all 7 activation options: none, relu, sigmoid, tanh, softmax, elu, selu
- [ ] Dense node property panel shows the same activation options for consistency
- [ ] Default activation for Conv2D is "none"
- [ ] Selecting each activation option correctly updates the node data
- [ ] Backend correctly handles all activation values including "none" (mapped to `None` in Keras)
- [ ] A Conv2D layer with each activation type validates successfully in a model
- [ ] The Conv2D node on the canvas displays the selected activation name

## Resources

- `tensormap-frontend/src/components/DragAndDropCanvas/NodePropertiesPanel.jsx` — property editor
- `tensormap-frontend/src/components/DragAndDropCanvas/CustomNodes/ConvNode/ConvNode.jsx` — Conv2D node display
- `tensormap-frontend/src/components/DragAndDropCanvas/Canvas.jsx` — default params
- `tensormap-backend/app/services/model_generation.py` — activation handling in `_build_layer()`
- [Keras activation functions](https://www.tensorflow.org/api_docs/python/tf/keras/activations)

### PR DESCRIPTION
**Pull request by kartikeyg0104**
Closes #129

## Summary

Expands the activation function dropdown for the Conv2D node from the two hardcoded options (none, relu) to the full set of commonly used Keras activation functions: none, relu, sigmoid, tanh, softmax, elu, selu.

For consistency, the Dense node dropdown is unified to use the same list. A single shared `ACTIVATIONS` constant replaces the two separate `denseActivations` and `convActivations` arrays (DRY). The backend already maps `"none"` → `"linear"` for Conv2D; the same mapping is now applied to Dense so both layer types behave correctly when `"none"` is selected.

### Root Cause
`convActivations` in `NodePropertiesPanel.jsx` only contained `["none", "relu"]`. The backend had no issues  it passes the activation string directly to `tf.keras.layers.Conv2D(activation=…)`, which natively supports all standard Keras names.

### Frontend
- **`NodePropertiesPanel.jsx`** - Replaced separate `denseActivations` and `convActivations` arrays with a single `ACTIVATIONS` constant containing all 7 options. Both the Dense and Conv2D `<Select>` dropdowns now render from this shared list.

### Backend
- **`model_generation.py`** - Added `"none"` → `"linear"` mapping for the `customdense` branch in `_build_layer()` to match the existing Conv2D handling.

## How to Test

```bash
# Frontend - start dev server and verify dropdowns
cd tensormap-frontend
npm install
npm run dev
# Open canvas, add a Conv2D node, open properties panel - confirm 7 activation options appear
# Repeat for Dense node

# Backend - run existing tests
cd tensormap-backend
uv run pytest -v
```

### CODE REVIEW

**Review by ivantha — APPROVED** [review]
Lgtm 👍🏼 Please rebase.
